### PR TITLE
Add `order` argument to` empty_like`,` zeros_like`, etc.

### LINKED ISSUE #1654: Feature: preserve memory layout order in empty_like, etc?
One thing that was surprising to me on using `cupy.zeros_like`, `cupy.empty_like` or `cupy.ones_like` were that when the input array was Fortran-ordered, the output is C-ordered. This differs from the NumPy behavior which preserves order. I see that it the docstring does state that "this function does not support order and subok options", so I suppose this is to be expected.

Would the CuPy devs be open to a PR that would preserve order 'C' or 'F' for contiguous arrays in these functions?. I think supporting the `order` parameter for values of 'C', 'F' and 'A' should be easy to do and would be willing to make a PR for that. (The underlying `cupy.zeros`, etc. already have the necessary `order` argument).

p.s. 
Congrats on the CuPy project! It is very nice and with the recently added support for custom kernels, I think I would prefer it going forward to PyCUDA in many cases due to the more complete NumPy API coverage.

### PR DESCRIPTION
**Pull request by grlee77**
closes #1654

### CODE REVIEW

File: cupy/creation/basic.py [hunk 1]
```diff
@@ -20,14 +20,67 @@ def empty(shape, dtype=float, order='C'):
     return cupy.ndarray(shape, dtype, order=order)
 
 
-def empty_like(a, dtype=None):
+def _new_like_order_and_strides(a, dtype, order):
```
[the remark this answers is not in the document]
Comment (grlee77) [reply cupy/creation/basic.py]: Thanks for pointing out that code! I refactored the Cython code slightly and then used the refactored functions to replace the majority of the code in `_new_like_order_and_strides`.

There is still a subtle difference in numpy vs. cupy if you pass a non-string argument to empty_like, etc.  On my system with numpy 1.15.1, this raises a warning:
```"DeprecationWarning: Non-string object detected for the array ordering. Please pass in 'C', 'F', 'A', or 'K' instead"```

whereas this cupy implementation would instead raise an error:
```AttributeError: 'int' object has no attribute 'upper'```

What do you recommend in this case? (we could do a try/except around the call to `upper()` and raise a similar warning, but I'm not sure it is worth it or when the warning may change to an error).

[the remark this answers is not in the document]
Comment (niboshi) [reply cupy/creation/basic.py]: Sorry for late reply. I think an error is more appropriate because we are not deprecating it.

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
Thank you very much!

**Pull request merged**